src/vma/proto/dst_entry.cpp

```diff
@@ -70,7 +70,7 @@ dst_entry::~dst_entry()
 	}
 
 	if (m_p_rt_entry) {
-		g_p_route_table_mgr->unregister_observer(route_rule_table_key(m_dst_ip.get_in_addr(), m_bound_ip ? m_bound_ip : m_so_bindtodevice_ip, m_tos), this);
+		g_p_route_table_mgr->unregister_observer(route_rule_table_key(m_dst_ip.get_in_addr(), m_src_ip, m_tos), this);
```

Comment:
Lets go over the changing of src_ip selection F2F - in particular the changing of m_bound_ip an m_so_bindtodevice_ip usage.